### PR TITLE
config: pass tower manager pointer in to PopulateDependencies

### LINKED ISSUE #8402: [bug]: Segmentation fault while trying to run lncli wtclient towers
### Background

I'm getting following dump while trying to run `lncli --no-macaroons wtclient towers`
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x103e79628]

goroutine 731 [running]:
github.com/lightningnetwork/lnd/watchtower/wtclient.(*Manager).RegisteredTowers(0x0, {0x0, 0x0, 0x0})
        /Users/animesh/Desktop/bitcoin/lnd/watchtower/wtclient/manager.go:458 +0x48
github.com/lightningnetwork/lnd/lnrpc/wtclientrpc.(*WatchtowerClient).ListTowers(0x1400073ade0, {0x1047a5858, 0x14001a85050}, 0x14001a85080)
        /Users/animesh/Desktop/bitcoin/lnd/lnrpc/wtclientrpc/wtclient.go:266 +0x148
github.com/lightningnetwork/lnd/lnrpc/wtclientrpc._WatchtowerClient_ListTowers_Handler.func1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080})
        /Users/animesh/Desktop/bitcoin/lnd/lnrpc/wtclientrpc/wtclient_grpc.pb.go:224 +0xb0
github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).middlewareUnaryServerInterceptor.func1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080}, 0x1400074aa80, 0x140006bddb8)
        /Users/animesh/Desktop/bitcoin/lnd/rpcperms/interceptor.go:819 +0x45c
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080})
        /Users/animesh/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x78
github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).MacaroonUnaryServerInterceptor.func1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080}, 0x1400074aa80, 0x1400074aaa0)
        /Users/animesh/Desktop/bitcoin/lnd/rpcperms/interceptor.go:689 +0xc8
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080})
        /Users/animesh/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x78
github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).rpcStateUnaryServerInterceptor.func1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080}, 0x1400074aa80, 0x1400074aac0)
        /Users/animesh/Desktop/bitcoin/lnd/rpcperms/interceptor.go:781 +0x17c
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080})
        /Users/animesh/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x78
github.com/lightningnetwork/lnd/rpcperms.errorLogUnaryServerInterceptor.func1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080}, 0x1400074aa80, 0x1400074aae0)
        /Users/animesh/Desktop/bitcoin/lnd/rpcperms/interceptor.go:605 +0x74
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080})
        /Users/animesh/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x78
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x1047a5858, 0x14001a85050}, {0x10467b700, 0x14001a85080}, 0x1400074aa80, 0x140006bddb8)
        /Users/animesh/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0x140
github.com/lightningnetwork/lnd/lnrpc/wtclientrpc._WatchtowerClient_ListTowers_Handler({0x1046d8ba0, 0x1400014fc30}, {0x1047a5858, 0x14001a85050}, 0x140003cd110, 0x140003ea420)
        /Users/animesh/Desktop/bitcoin/lnd/lnrpc/wtclientrpc/wtclient_grpc.pb.go:226 +0x1d8
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000402000, {0x1047ad600, 0x140007add40}, 0x14000498a20, 0x140003ea8a0, 0x10523df30, 0x0)
        /Users/animesh/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:1335 +0x1214
google.golang.org/grpc.(*Server).handleStream(0x14000402000, {0x1047ad600, 0x140007add40}, 0x14000498a20, 0x0)
        /Users/animesh/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:1712 +0x71c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        /Users/animesh/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:947 +0x168
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 730
        /Users/animesh/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:958 +0x1fc
```

### Your environment

* version of `lnd` : `Version: 0.17.0-beta commit=fn/v1.0.1-35-gba4021cad`
* which operating system (`uname -a` on *Nix) : `MacOS 13.0.1`
* version of `btcd`, `bitcoind`, or other backend : `Bitcoin Core version v26.99.0-d1717b2c95c4 (release build)`
* any other relevant environment details
[bitcoin_conf.txt](https://github.com/lightningnetwork/lnd/files/13988334/bitcoin_conf.txt)
[lnd_conf.txt](https://github.com/lightningnetwork/lnd/files/13988335/lnd_conf.txt)

### Steps to reproduce

CMD to start LND: `./lnd-debug --rpclisten=localhost:10009 --listen=localhost:10011 --restlisten=localhost:8001 --configfile=/Users/animesh/Desktop/bitcoin/config_files/lnd.conf`
CMD to start bitcoind: `./src/bitcoind -reindex -prune=600 -conf=/Users/animesh/Desktop/bitcoin/config_files/bitcoin.conf -allowignoredconf=1 `

RUN: `lncli --no-macaroons wtclient towers`
LND crashes

NOTE: I was using regtest

### Expected behaviour

LND should be able to handle scenarios gracefully without crashing, at worst the end user should see some informative error.

### Actual behaviour

LND crashes

### PR DESCRIPTION
**Pull request by ellemouton**
In this commit, we let the explicit wtclient.Manager struct be passed in to PopulateDependencies instead of the tower client interface. We need to do this since we do allow the tower client interface to be nil if the client is not active. So to avoid the golang gotcha where the interface value will be seen as not nil even though the underlying value is nil, we pass in the explicit pointer instead.

Basically what would happen is that `nil` would be passed in for `towerClientMgr` but since it is an interface, the `towerClientMgr != nil` check would still evaluate to true meaning that the tower config `Active` would get set to true even though now tower manager has been created.

(Note: adding the `no-changelog` tag since this bug hasnt been included in a release yet). 

Fixes #8402

### CODE REVIEW

**Comment by anibilthare** [comment]
Screenshot 2024-01-19 at 5 36 49 PM
Screenshot 2024-01-19 at 5 36 58 PM


Not seeing the crash anymore! Thanks for the quick fix @ellemouton !

**Comment by ellemouton** [comment]
Thanks for testing @anibilthare !

**Review by yyforyongyu — APPROVED** [review]
LGTM ⚡

**Review by guggero — APPROVED** [review]
